std/array.d

```diff
@@ -2955,7 +2955,7 @@ if (isDynamicArray!A)
      * Params:
      *     newCapacity = the capacity the `Appender` should have
      */
-    void reserve(size_t newCapacity) @safe pure nothrow
+    void reserve(size_t newCapacity) @safe nothrow
```

Comment:
If the issue is that whether the function works as `pure` or not based on the template argument, then the function shouldn't be marked with `@safe` or `nothrow` either. All such attributes should be inferred. So, you should remove all attributes from both functions and ensure that they're tested appropriately in the unit tests, not just `pure`.

Reply:
I think this require a bigger decision because probably we should remove all attributes from a lot of places. This is just a fix for bug 13300.